src/stan/math/prim/mat/prob/multi_normal_rng.hpp

```diff
@@ -20,10 +20,10 @@ namespace stan {
   namespace prob {
     using Eigen::Dynamic;
 
-    template <class RNG>
+    template <class RNG, typename T1, typename T2>
```

Comment:
Pretty sure you can with enable_if.  For example,

  template<typename DerivedA, typename DerivedB>
  inline typename std::enable_if<
    is_var<typename Eigen::MatrixBase<DerivedA>::Scalar>::value &&
    is_var<typename Eigen::MatrixBase<DerivedB>::Scalar>::value &&
    std::is_same<typename Eigen::MatrixBase<DerivedA>::Scalar,
                 typename Eigen::MatrixBase<DerivedB>::Scalar>::value,
    typename Eigen::Matrix<typename Eigen::MatrixBase<DerivedA>::Scalar,
             Eigen::Dynamic, Eigen::Dynamic>
    >::type
  multiply(const Eigen::MatrixBase<DerivedA>& M1,
           const Eigen::MatrixBase<DerivedB>& M2) {�}


On Mar 12, 2015, at 9:47 AM, Joshua Pritikin <notifications@github.com> wrote:

> In src/stan/math/prim/mat/prob/multi_normal_rng.hpp:
> 
> > @@ -20,10 +20,10 @@ namespace stan {
> >    namespace prob {
> >      using Eigen::Dynamic;
> >  
> > -    template <class RNG>
> > +    template <class RNG, typename T1, typename T2>
> 
> But T1 and T2 are not scalar types to begin with. You can't constrain the scalar type using the template when you are using Eigen::MatrixBase<>
> 
> �
> Reply to this email directly or view it on GitHub.
>